### PR TITLE
Run kustomize edit fix on config directory to remove deprecated fields

### PR DESCRIPTION
**Pull request by kschoche**
Updates the kustomization configuration throughout the config directory to no longer use deprecated fields:

```
TestVaultDynamicSecret 2023-04-27T11:04:08-05:00 logger.go:66: # Warning: 'bases' is deprecated. Please use 'resources' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
TestVaultDynamicSecret 2023-04-27T11:04:08-05:00 logger.go:66: # Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
TestVaultDynamicSecret 2023-04-27T11:04:08-05:00 logger.go:66: # Warning: 'bases' is deprecated. Please use 'resources' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
TestVaultDynamicSecret 2023-04-27T11:04:08-05:00 logger.go:66: # Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
```

With these fixes, the warnings go away and tests still pass and demo still deploys.

### CODE REVIEW

File: config/rbac/kustomization.yaml [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-resources:
 # All RBAC will be applied under this service account in
 # the deployment namespace. You may comment out this resource
 # if your manager will use a service account that exists at
 # runtime. Be sure to update RoleBinding and ClusterRoleBinding
 # subjects if changing service account names.
+# Comment the following 4 lines if you want to disable
+# the auth proxy (https://github.com/brancz/kube-rbac-proxy)
+# which protects your /metrics endpoint.
+resources:
```
Comment (benashz) [inline config/rbac/kustomization.yaml:12]: I think this comment belongs with the `auth_proxy*` bits below.

File: config/samples/kustomization.yaml [hunk 2]
```diff
@@ -9,4 +9,5 @@ resources:
 - secrets_v1alpha1_vaultauth.yaml
 - secrets_v1alpha1_vaultconnection.yaml
 - secrets_v1alpha1_vaultdynamicsecret.yaml
-#+kubebuilder:scaffold:manifestskustomizesamples
```
Comment (benashz) [inline config/samples/kustomization.yaml:12]: Do we still need the kubebuilder marker?

File: config/scorecard/kustomization.yaml [hunk 3]
```diff
@@ -3,17 +3,18 @@
...
     kind: Configuration
     name: config
+    version: v1alpha3
 - path: patches/olm.config.yaml
   target:
     group: scorecard.operatorframework.io
-    version: v1alpha3
     kind: Configuration
     name: config
-#+kubebuilder:scaffold:patchesJson6902
```
Comment (benashz) [inline config/scorecard/kustomization.yaml:19]: Do we still need the kubebuilder marker?

File: config/manager/kustomization.yaml [hunk 4]
```diff
@@ -16,4 +16,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: hashicorp/vault-secrets-operator
-  newTag: 0.1.0-beta
```
Comment (benashz) [inline config/manager/kustomization.yaml:19]: This change keeps popping up in other PRs. I think it is a by product of running kustomize.

**Review by benashz — COMMENTED** [review]
Had a few questions/comments.

File: config/manager/kustomization.yaml (hunk 4 above)
Comment (kschoche) [reply config/manager/kustomization.yaml:19]: This one is so annoying.. idk what to do with it, nearly every commit is getting this when we're working on CRDs at this point.

File: config/scorecard/kustomization.yaml (hunk 3 above)
Comment (kschoche) [reply config/scorecard/kustomization.yaml:19]: Good catch,  both this one and the other one you mentioned were auto-removed by `kustomize edit fix` so I assumed it was correct, let me dig into it on monday!

**Comment by kschoche** [comment]
Closing until there's time to come back to this.

**Pull request closed without merging**